presto-main/src/test/java/io/prestosql/sql/planner/assertions/BasePlanTest.java

```diff
@@ -87,6 +92,13 @@ private static LocalQueryRunner createQueryRunner(Map<String, String> sessionPro
     public final void initPlanTest()
     {
         queryRunner = queryRunnerSupplier.get();
+        List<CatalogContext> catalogs = setupCatalogs();
```

Comment:
Why not to inject query runner supplier with `BasePlanTest(LocalQueryRunnerSupplier supplier)`? Then you could create query runner with any catalogs you like. No need to to pass `CatalogContext` etc.

Reply:
@kokosing If we do that, then the class extending `BasePlanTest` will need to construct this supplier in the constructor, which makes things a bit rigid because then it can only be created through a static method. For example, if we want to add something like this in hive, it requires setting up directories and cleaning them up at the end. If we insist on using queryrunner, it will require everything to be static and initialized in the constructor. Do we want to force that? https://github.com/prestosql/presto/pull/1720/commits/fc677338cb8a62900b43b647aff8f2e79f5f4175#diff-a5282cfb6f3974b1d9e2fd6f464302efR70

Reply:
You can use something like: `io.prestosql.testing.AbstractTestQueryFramework#createQueryRunner`